File: src/main/java/com/skyspace666/service/LoyaltyMemberService.java
package com.skyspace666.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace666.domain.LoyaltyMember;
import com.skyspace666.dto.LoyaltyMemberDTO;
import com.skyspace666.dto.LoyaltyMemberSearchDTO;
import com.skyspace666.dto.LoyaltyMemberPageDTO;
import com.skyspace666.dto.LoyaltyMemberConvertCriteriaDTO;
import com.skyspace666.service.GenericService;
import com.skyspace666.dto.common.RequestDTO;
import com.skyspace666.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface LoyaltyMemberService extends GenericService<LoyaltyMember, Integer> {

	List<LoyaltyMember> findAll();

	ResultDTO addLoyaltyMember(LoyaltyMemberDTO loyaltyMemberDTO, RequestDTO requestDTO);

	ResultDTO updateLoyaltyMember(LoyaltyMemberDTO loyaltyMemberDTO, RequestDTO requestDTO);

    Page<LoyaltyMember> getAllLoyaltyMembers(Pageable pageable);

    Page<LoyaltyMember> getAllLoyaltyMembers(Specification<LoyaltyMember> spec, Pageable pageable);

	ResponseEntity<LoyaltyMemberPageDTO> getLoyaltyMembers(LoyaltyMemberSearchDTO loyaltyMemberSearchDTO);
	
	List<LoyaltyMemberDTO> convertLoyaltyMembersToLoyaltyMemberDTOs(List<LoyaltyMember> loyaltyMembers, LoyaltyMemberConvertCriteriaDTO convertCriteria);

	LoyaltyMemberDTO getLoyaltyMemberDTOById(Integer loyaltyMemberId);







}

File: skyspace666_ui/src/services/LoyaltyMemberService.js
import http from "../http-common"; 

class LoyaltyMemberService {
  getAllLoyaltyMembers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/loyaltyMember/loyaltyMembers`, searchDTO);
  }

  get(loyaltyMemberId) {
    return this.getRequest(`/loyaltyMember/${loyaltyMemberId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/loyaltyMember?field=${matchData}`, null);
  }

  addLoyaltyMember(data) {
    return http.post("/loyaltyMember/addLoyaltyMember", data);
  }

  update(data) {
  	return http.post("/loyaltyMember/updateLoyaltyMember", data);
  }
  
  uploadImage(data,loyaltyMemberId) {
  	return http.postForm("/loyaltyMember/uploadImage/"+loyaltyMemberId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new LoyaltyMemberService();
